src/demo/renderer.js

```diff
@@ -291,12 +316,16 @@ const drawMovingFish = state => {
       fish.getResult() ? fish.getResult().predictedClassId : false
     );
 
+    let drawPrediction = false;
     if (state.currentMode === Modes.Predicting) {
       if (fish.getResult()) {
         const midScreenX =
           constants.canvasWidth / 2 - constants.fishCanvasWidth / 2;
-        if (Math.abs(midScreenX - x) <= 50) {
+        drawPrediction = x >= midScreenX;
+
+        if (drawPrediction && x - midScreenX <= 50) {
```

Comment:
I wonder if we should hoist `x - midScreenX <= 50` into a variable as well just so we that we can name this additional condition...

Reply:
good call, adding that now